fabric-rendering-v1/src/client/java/net/fabricmc/fabric/mixin/client/rendering/WorldRendererMixin.java

```diff
@@ -86,6 +85,12 @@ private void afterTerrainSolid(CallbackInfo ci) {
 		WorldRenderEvents.BEFORE_ENTITIES.invoker().beforeEntities(context);
 	}
 
+	@ModifyExpressionValue(method = "render", at = @At(value = "NEW", target = "net/minecraft/client/util/math/MatrixStack"))
+	private MatrixStack setMatrixStack(MatrixStack matrixStack) {
+		context.setMatrixStack(matrixStack);
```

Comment:
The context object is re-used, so you need to set it back to null in context.prepare

Reply:
Done, but other objects such as `frustum` and block outline stuff such as `entity` and `blockState` are also not reset.